File: js/packages/app/src/__tests__/features/object-references/external/polywrap/object-references-example/src/polywrap/external/module/WrapModule.ts
import { IExternalWrapInstance } from "@nerfzael/reim-wrap-js";
//import { ImportBindings } from "./ImportBindings";

//
//
//import { create as createTestReceiveReference } from "../global-functions/testReceiveReference";
//
//
//
//import { create as createTestInvokeExternalGlobalFunction } from "../global-functions/testInvokeExternalGlobalFunction";
//
//
//
//import { create as createTestInvokeExternalStaticMethod } from "../global-functions/testInvokeExternalStaticMethod";
//
//
//
//import { create as createTestInvokeExternalInstanceMethod } from "../global-functions/testInvokeExternalInstanceMethod";
//
//
//
//
//
//
//import { create as createTestObjectGetter } from "../classes/TestObjectGetter";
//
//
//
//

export class WrapModule {
  static wrapInstance: IExternalWrapInstance | null;

  static connect(instance: IExternalWrapInstance): void {
    WrapModule.wrapInstance = instance;
  }

  //static import(instance: IExternalWrapInstance): ImportBindings {
  //  return new ImportBindings(
  //    
  //    
  //    createTestReceiveReference(instance),
  //    
  //    
  //    
  //    createTestInvokeExternalGlobalFunction(instance),
  //    
  //    
  //    
  //    createTestInvokeExternalStaticMethod(instance),
  //    
  //    
  //    
  //    createTestInvokeExternalInstanceMethod(instance),
  //    
  //    
  //    
  //    
  //    
  //    
  //    createTestObjectGetter(instance),
  //    
  //    
  //    
  //    
  //  );
  //}
}

File: rust/app/src/polywrap/external_references/src/polywrap/external/module/WrapModule.rs
use reim_wrap::{ ExternalWrapModule };
use ImportBindings;

use external::global_functions::testReceiveReference::{ create as createTestReceiveReference };
use external::global_functions::testInvokeExternalGlobalFunction::{ create as createTestInvokeExternalGlobalFunction };
use external::global_functions::testInvokeExternalStaticMethod::{ create as createTestInvokeExternalStaticMethod };
use external::global_functions::testInvokeExternalInstanceMethod::{ create as createTestInvokeExternalInstanceMethod };

use external::classes::TestObjectGetter::{ create as createTestObjectGetter };


struct WrapModule {}

impl WrapModule {
  static wrapInstance: Option<Box<dyn ExternalWrapModule>>;

  pub fn connect(instance: dyn ExternalWrapModule) {
    WrapModule.wrapInstance = Box::new(instance);
  }

  pub fn import(instance: dyn ExternalWrapModule) -> ImportBindings {
    ImportBindings.new(
      
      createTestReceiveReference(instance),
            
      createTestInvokeExternalGlobalFunction(instance),
            
      createTestInvokeExternalStaticMethod(instance),
            
      createTestInvokeExternalInstanceMethod(instance),
                  
      
      createTestObjectGetter(instance),
                  
    )
  }
}
